In [1]:
import pandas as pd 

#### Import the libraries pandas,matplot.pyplot an seaborn from python,in ordr tp perform coulmn operations on dataset

In [2]:
sample=pd.read_csv("D:/flipkart_smartphones.csv")

In [3]:
sample

brand             model               colour  original_price  \
0      VIVO      VIVO T1 44W            Starry Sky           19990   
1     APPLE  APPLE IPHONE 11                 White           48900   
2      VIVO      VIVO T1 44W       Midnight Galaxy           20990   
3    XIAOMI       POCO M4 5G           Power Black           15999   
4    XIAOMI         REDMI 10       Caribbean Green           14999   
..      ...               ...                  ...             ...   
831  REALME  REALME GT NEO 2              NEO Blue           38999   
832  REALME  REALME GT NEO 2             NEO Black           38999   
833  REALME  REALME GT NEO 2             NEO Black           34999   
834  REALME   REALME X50 PRO              Rust Red           17999   
835  XIAOMI      POCO M2 PRO   Two Shades of Black           17999   

     discounted_price  ratings  rating_count  reviews  memory  storage  \
0               14499      4.5         87331     6044     4.0    128.0   
1               47199      4.6        184191    10818     NaN    128.0   
2               15999      4.4         51365     3750     6.0    128.0   
3               11999      4.2         53448     4185     4.0     64.0   
4                9299      NaN        187787    12084     4.0     64.0   
..                ...      ...           ...      ...     ...      ...   
831             35999      4.4          4430      628     NaN    256.0   
832             35999      4.4          4430      628     NaN    256.0   
833             31999      4.4         18194     2478     8.0    128.0   
834             41999      4.4          8689     1232     8.0    128.0   
835             17999      4.4        350744    33202     6.0     64.0   

                    processor               rear_camera      front_camera  \
0     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
1               A Bionic Chip              12MP + 12MP               12MP   
2     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
3      Mediatek Dimensity 700               50MP + 2MP                8MP   
4     Qualcomm Snapdragon 680               50MP + 2MP                5MP   
..                        ...                       ...               ...   
831   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
832   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
833   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
834                       NaN  64MP + 12MP + 8MP + 2MP    32MP + 8MP Dual   
835  Qualcomm Snapdragon 720G   48MP + 8MP + 5MP + 2MP               16MP   

     display_size  battery_capacity     battery_type  
0           16.36            5000.0          Lithium  
1           15.49               NaN              NaN  
2           16.36            5000.0          Lithium  
3           16.71            5000.0  Lithium Polymer  
4           17.02            6000.0  Lithium Polymer  
..            ...               ...              ...  
831         16.81            5000.0              NaN  
832         16.81            5000.0              NaN  
833         16.81            5000.0              NaN  
834         16.36            4200.0              NaN  
835         16.94            5000.0          Lithium  

[836 rows x 16 columns]

In [4]:
sample.columns

Index(['brand', 'model', 'colour', 'original_price', 'discounted_price',
       'ratings', 'rating_count', 'reviews', 'memory', 'storage', 'processor',
       'rear_camera', 'front_camera', 'display_size', 'battery_capacity',
       'battery_type'],
      dtype='object')

#### Displaying the variables names in the data set

In [5]:
sample['total']=sample['original_price']+sample['discounted_price']

In [6]:
sample

brand             model               colour  original_price  \
0      VIVO      VIVO T1 44W            Starry Sky           19990   
1     APPLE  APPLE IPHONE 11                 White           48900   
2      VIVO      VIVO T1 44W       Midnight Galaxy           20990   
3    XIAOMI       POCO M4 5G           Power Black           15999   
4    XIAOMI         REDMI 10       Caribbean Green           14999   
..      ...               ...                  ...             ...   
831  REALME  REALME GT NEO 2              NEO Blue           38999   
832  REALME  REALME GT NEO 2             NEO Black           38999   
833  REALME  REALME GT NEO 2             NEO Black           34999   
834  REALME   REALME X50 PRO              Rust Red           17999   
835  XIAOMI      POCO M2 PRO   Two Shades of Black           17999   

     discounted_price  ratings  rating_count  reviews  memory  storage  \
0               14499      4.5         87331     6044     4.0    128.0   
1               47199      4.6        184191    10818     NaN    128.0   
2               15999      4.4         51365     3750     6.0    128.0   
3               11999      4.2         53448     4185     4.0     64.0   
4                9299      NaN        187787    12084     4.0     64.0   
..                ...      ...           ...      ...     ...      ...   
831             35999      4.4          4430      628     NaN    256.0   
832             35999      4.4          4430      628     NaN    256.0   
833             31999      4.4         18194     2478     8.0    128.0   
834             41999      4.4          8689     1232     8.0    128.0   
835             17999      4.4        350744    33202     6.0     64.0   

                    processor               rear_camera      front_camera  \
0     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
1               A Bionic Chip              12MP + 12MP               12MP   
2     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
3      Mediatek Dimensity 700               50MP + 2MP                8MP   
4     Qualcomm Snapdragon 680               50MP + 2MP                5MP   
..                        ...                       ...               ...   
831   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
832   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
833   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
834                       NaN  64MP + 12MP + 8MP + 2MP    32MP + 8MP Dual   
835  Qualcomm Snapdragon 720G   48MP + 8MP + 5MP + 2MP               16MP   

     display_size  battery_capacity     battery_type  total  
0           16.36            5000.0          Lithium  34489  
1           15.49               NaN              NaN  96099  
2           16.36            5000.0          Lithium  36989  
3           16.71            5000.0  Lithium Polymer  27998  
4           17.02            6000.0  Lithium Polymer  24298  
..            ...               ...              ...    ...  
831         16.81            5000.0              NaN  74998  
832         16.81            5000.0              NaN  74998  
833         16.81            5000.0              NaN  66998  
834         16.36            4200.0              NaN  59998  
835         16.94            5000.0          Lithium  35998  

[836 rows x 17 columns]

#### Added a new variables in the dataset bycombining the variables

In [7]:
sample['low']=sample['rating_count']/sample['reviews']

In [8]:
sample

brand             model               colour  original_price  \
0      VIVO      VIVO T1 44W            Starry Sky           19990   
1     APPLE  APPLE IPHONE 11                 White           48900   
2      VIVO      VIVO T1 44W       Midnight Galaxy           20990   
3    XIAOMI       POCO M4 5G           Power Black           15999   
4    XIAOMI         REDMI 10       Caribbean Green           14999   
..      ...               ...                  ...             ...   
831  REALME  REALME GT NEO 2              NEO Blue           38999   
832  REALME  REALME GT NEO 2             NEO Black           38999   
833  REALME  REALME GT NEO 2             NEO Black           34999   
834  REALME   REALME X50 PRO              Rust Red           17999   
835  XIAOMI      POCO M2 PRO   Two Shades of Black           17999   

     discounted_price  ratings  rating_count  reviews  memory  storage  \
0               14499      4.5         87331     6044     4.0    128.0   
1               47199      4.6        184191    10818     NaN    128.0   
2               15999      4.4         51365     3750     6.0    128.0   
3               11999      4.2         53448     4185     4.0     64.0   
4                9299      NaN        187787    12084     4.0     64.0   
..                ...      ...           ...      ...     ...      ...   
831             35999      4.4          4430      628     NaN    256.0   
832             35999      4.4          4430      628     NaN    256.0   
833             31999      4.4         18194     2478     8.0    128.0   
834             41999      4.4          8689     1232     8.0    128.0   
835             17999      4.4        350744    33202     6.0     64.0   

                    processor               rear_camera      front_camera  \
0     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
1               A Bionic Chip              12MP + 12MP               12MP   
2     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
3      Mediatek Dimensity 700               50MP + 2MP                8MP   
4     Qualcomm Snapdragon 680               50MP + 2MP                5MP   
..                        ...                       ...               ...   
831   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
832   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
833   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
834                       NaN  64MP + 12MP + 8MP + 2MP    32MP + 8MP Dual   
835  Qualcomm Snapdragon 720G   48MP + 8MP + 5MP + 2MP               16MP   

     display_size  battery_capacity     battery_type  total        low  
0           16.36            5000.0          Lithium  34489  14.449206  
1           15.49               NaN              NaN  96099  17.026345  
2           16.36            5000.0          Lithium  36989  13.697333  
3           16.71            5000.0  Lithium Polymer  27998  12.771326  
4           17.02            6000.0  Lithium Polymer  24298  15.540136  
..            ...               ...              ...    ...        ...  
831         16.81            5000.0              NaN  74998   7.054140  
832         16.81            5000.0              NaN  74998   7.054140  
833         16.81            5000.0              NaN  66998   7.342211  
834         16.36            4200.0              NaN  59998   7.052760  
835         16.94            5000.0          Lithium  35998  10.563942  

[836 rows x 18 columns]

In [9]:
sample.rename(columns={'reviews':'review'})

brand             model               colour  original_price  \
0      VIVO      VIVO T1 44W            Starry Sky           19990   
1     APPLE  APPLE IPHONE 11                 White           48900   
2      VIVO      VIVO T1 44W       Midnight Galaxy           20990   
3    XIAOMI       POCO M4 5G           Power Black           15999   
4    XIAOMI         REDMI 10       Caribbean Green           14999   
..      ...               ...                  ...             ...   
831  REALME  REALME GT NEO 2              NEO Blue           38999   
832  REALME  REALME GT NEO 2             NEO Black           38999   
833  REALME  REALME GT NEO 2             NEO Black           34999   
834  REALME   REALME X50 PRO              Rust Red           17999   
835  XIAOMI      POCO M2 PRO   Two Shades of Black           17999   

     discounted_price  ratings  rating_count  review  memory  storage  \
0               14499      4.5         87331    6044     4.0    128.0   
1               47199      4.6        184191   10818     NaN    128.0   
2               15999      4.4         51365    3750     6.0    128.0   
3               11999      4.2         53448    4185     4.0     64.0   
4                9299      NaN        187787   12084     4.0     64.0   
..                ...      ...           ...     ...     ...      ...   
831             35999      4.4          4430     628     NaN    256.0   
832             35999      4.4          4430     628     NaN    256.0   
833             31999      4.4         18194    2478     8.0    128.0   
834             41999      4.4          8689    1232     8.0    128.0   
835             17999      4.4        350744   33202     6.0     64.0   

                    processor               rear_camera      front_camera  \
0     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
1               A Bionic Chip              12MP + 12MP               12MP   
2     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
3      Mediatek Dimensity 700               50MP + 2MP                8MP   
4     Qualcomm Snapdragon 680               50MP + 2MP                5MP   
..                        ...                       ...               ...   
831   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
832   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
833   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
834                       NaN  64MP + 12MP + 8MP + 2MP    32MP + 8MP Dual   
835  Qualcomm Snapdragon 720G   48MP + 8MP + 5MP + 2MP               16MP   

     display_size  battery_capacity     battery_type  total        low  
0           16.36            5000.0          Lithium  34489  14.449206  
1           15.49               NaN              NaN  96099  17.026345  
2           16.36            5000.0          Lithium  36989  13.697333  
3           16.71            5000.0  Lithium Polymer  27998  12.771326  
4           17.02            6000.0  Lithium Polymer  24298  15.540136  
..            ...               ...              ...    ...        ...  
831         16.81            5000.0              NaN  74998   7.054140  
832         16.81            5000.0              NaN  74998   7.054140  
833         16.81            5000.0              NaN  66998   7.342211  
834         16.36            4200.0              NaN  59998   7.052760  
835         16.94            5000.0          Lithium  35998  10.563942  

[836 rows x 18 columns]

#### Renamed a reviews variables as review

In [10]:
sample.rename(columns={'brand':'brands','model':'models'})

brands            models               colour  original_price  \
0      VIVO      VIVO T1 44W            Starry Sky           19990   
1     APPLE  APPLE IPHONE 11                 White           48900   
2      VIVO      VIVO T1 44W       Midnight Galaxy           20990   
3    XIAOMI       POCO M4 5G           Power Black           15999   
4    XIAOMI         REDMI 10       Caribbean Green           14999   
..      ...               ...                  ...             ...   
831  REALME  REALME GT NEO 2              NEO Blue           38999   
832  REALME  REALME GT NEO 2             NEO Black           38999   
833  REALME  REALME GT NEO 2             NEO Black           34999   
834  REALME   REALME X50 PRO              Rust Red           17999   
835  XIAOMI      POCO M2 PRO   Two Shades of Black           17999   

     discounted_price  ratings  rating_count  reviews  memory  storage  \
0               14499      4.5         87331     6044     4.0    128.0   
1               47199      4.6        184191    10818     NaN    128.0   
2               15999      4.4         51365     3750     6.0    128.0   
3               11999      4.2         53448     4185     4.0     64.0   
4                9299      NaN        187787    12084     4.0     64.0   
..                ...      ...           ...      ...     ...      ...   
831             35999      4.4          4430      628     NaN    256.0   
832             35999      4.4          4430      628     NaN    256.0   
833             31999      4.4         18194     2478     8.0    128.0   
834             41999      4.4          8689     1232     8.0    128.0   
835             17999      4.4        350744    33202     6.0     64.0   

                    processor               rear_camera      front_camera  \
0     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
1               A Bionic Chip              12MP + 12MP               12MP   
2     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
3      Mediatek Dimensity 700               50MP + 2MP                8MP   
4     Qualcomm Snapdragon 680               50MP + 2MP                5MP   
..                        ...                       ...               ...   
831   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
832   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
833   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
834                       NaN  64MP + 12MP + 8MP + 2MP    32MP + 8MP Dual   
835  Qualcomm Snapdragon 720G   48MP + 8MP + 5MP + 2MP               16MP   

     display_size  battery_capacity     battery_type  total        low  
0           16.36            5000.0          Lithium  34489  14.449206  
1           15.49               NaN              NaN  96099  17.026345  
2           16.36            5000.0          Lithium  36989  13.697333  
3           16.71            5000.0  Lithium Polymer  27998  12.771326  
4           17.02            6000.0  Lithium Polymer  24298  15.540136  
..            ...               ...              ...    ...        ...  
831         16.81            5000.0              NaN  74998   7.054140  
832         16.81            5000.0              NaN  74998   7.054140  
833         16.81            5000.0              NaN  66998   7.342211  
834         16.36            4200.0              NaN  59998   7.052760  
835         16.94            5000.0          Lithium  35998  10.563942  

[836 rows x 18 columns]

#### Renamed Two variables brand as brands and model as models

In [11]:
sample.drop(['rear_camera','front_camera'],axis=1)

brand             model               colour  original_price  \
0      VIVO      VIVO T1 44W            Starry Sky           19990   
1     APPLE  APPLE IPHONE 11                 White           48900   
2      VIVO      VIVO T1 44W       Midnight Galaxy           20990   
3    XIAOMI       POCO M4 5G           Power Black           15999   
4    XIAOMI         REDMI 10       Caribbean Green           14999   
..      ...               ...                  ...             ...   
831  REALME  REALME GT NEO 2              NEO Blue           38999   
832  REALME  REALME GT NEO 2             NEO Black           38999   
833  REALME  REALME GT NEO 2             NEO Black           34999   
834  REALME   REALME X50 PRO              Rust Red           17999   
835  XIAOMI      POCO M2 PRO   Two Shades of Black           17999   

     discounted_price  ratings  rating_count  reviews  memory  storage  \
0               14499      4.5         87331     6044     4.0    128.0   
1               47199      4.6        184191    10818     NaN    128.0   
2               15999      4.4         51365     3750     6.0    128.0   
3               11999      4.2         53448     4185     4.0     64.0   
4                9299      NaN        187787    12084     4.0     64.0   
..                ...      ...           ...      ...     ...      ...   
831             35999      4.4          4430      628     NaN    256.0   
832             35999      4.4          4430      628     NaN    256.0   
833             31999      4.4         18194     2478     8.0    128.0   
834             41999      4.4          8689     1232     8.0    128.0   
835             17999      4.4        350744    33202     6.0     64.0   

                    processor  display_size  battery_capacity  \
0     Qualcomm Snapdragon 680         16.36            5000.0   
1               A Bionic Chip         15.49               NaN   
2     Qualcomm Snapdragon 680         16.36            5000.0   
3      Mediatek Dimensity 700         16.71            5000.0   
4     Qualcomm Snapdragon 680         17.02            6000.0   
..                        ...           ...               ...   
831   Qualcomm Snapdragon 870         16.81            5000.0   
832   Qualcomm Snapdragon 870         16.81            5000.0   
833   Qualcomm Snapdragon 870         16.81            5000.0   
834                       NaN         16.36            4200.0   
835  Qualcomm Snapdragon 720G         16.94            5000.0   

        battery_type  total        low  
0            Lithium  34489  14.449206  
1                NaN  96099  17.026345  
2            Lithium  36989  13.697333  
3    Lithium Polymer  27998  12.771326  
4    Lithium Polymer  24298  15.540136  
..               ...    ...        ...  
831              NaN  74998   7.054140  
832              NaN  74998   7.054140  
833              NaN  66998   7.342211  
834              NaN  59998   7.052760  
835          Lithium  35998  10.563942  

[836 rows x 16 columns]

#### We have droped the two variables rear_camera and front_camera from the dataset

In [12]:
sample.columns=[col.upper()for col in sample.columns]

In [13]:
sample

BRAND             MODEL               COLOUR  ORIGINAL_PRICE  \
0      VIVO      VIVO T1 44W            Starry Sky           19990   
1     APPLE  APPLE IPHONE 11                 White           48900   
2      VIVO      VIVO T1 44W       Midnight Galaxy           20990   
3    XIAOMI       POCO M4 5G           Power Black           15999   
4    XIAOMI         REDMI 10       Caribbean Green           14999   
..      ...               ...                  ...             ...   
831  REALME  REALME GT NEO 2              NEO Blue           38999   
832  REALME  REALME GT NEO 2             NEO Black           38999   
833  REALME  REALME GT NEO 2             NEO Black           34999   
834  REALME   REALME X50 PRO              Rust Red           17999   
835  XIAOMI      POCO M2 PRO   Two Shades of Black           17999   

     DISCOUNTED_PRICE  RATINGS  RATING_COUNT  REVIEWS  MEMORY  STORAGE  \
0               14499      4.5         87331     6044     4.0    128.0   
1               47199      4.6        184191    10818     NaN    128.0   
2               15999      4.4         51365     3750     6.0    128.0   
3               11999      4.2         53448     4185     4.0     64.0   
4                9299      NaN        187787    12084     4.0     64.0   
..                ...      ...           ...      ...     ...      ...   
831             35999      4.4          4430      628     NaN    256.0   
832             35999      4.4          4430      628     NaN    256.0   
833             31999      4.4         18194     2478     8.0    128.0   
834             41999      4.4          8689     1232     8.0    128.0   
835             17999      4.4        350744    33202     6.0     64.0   

                    PROCESSOR               REAR_CAMERA      FRONT_CAMERA  \
0     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
1               A Bionic Chip              12MP + 12MP               12MP   
2     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
3      Mediatek Dimensity 700               50MP + 2MP                8MP   
4     Qualcomm Snapdragon 680               50MP + 2MP                5MP   
..                        ...                       ...               ...   
831   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
832   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
833   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
834                       NaN  64MP + 12MP + 8MP + 2MP    32MP + 8MP Dual   
835  Qualcomm Snapdragon 720G   48MP + 8MP + 5MP + 2MP               16MP   

     DISPLAY_SIZE  BATTERY_CAPACITY     BATTERY_TYPE  TOTAL        LOW  
0           16.36            5000.0          Lithium  34489  14.449206  
1           15.49               NaN              NaN  96099  17.026345  
2           16.36            5000.0          Lithium  36989  13.697333  
3           16.71            5000.0  Lithium Polymer  27998  12.771326  
4           17.02            6000.0  Lithium Polymer  24298  15.540136  
..            ...               ...              ...    ...        ...  
831         16.81            5000.0              NaN  74998   7.054140  
832         16.81            5000.0              NaN  74998   7.054140  
833         16.81            5000.0              NaN  66998   7.342211  
834         16.36            4200.0              NaN  59998   7.052760  
835         16.94            5000.0          Lithium  35998  10.563942  

[836 rows x 18 columns]

#### Displayed the variables name of the dataset in the uppercase

In [14]:
sample.columns=[col.lower()for col in sample.columns]

In [15]:
sample

brand             model               colour  original_price  \
0      VIVO      VIVO T1 44W            Starry Sky           19990   
1     APPLE  APPLE IPHONE 11                 White           48900   
2      VIVO      VIVO T1 44W       Midnight Galaxy           20990   
3    XIAOMI       POCO M4 5G           Power Black           15999   
4    XIAOMI         REDMI 10       Caribbean Green           14999   
..      ...               ...                  ...             ...   
831  REALME  REALME GT NEO 2              NEO Blue           38999   
832  REALME  REALME GT NEO 2             NEO Black           38999   
833  REALME  REALME GT NEO 2             NEO Black           34999   
834  REALME   REALME X50 PRO              Rust Red           17999   
835  XIAOMI      POCO M2 PRO   Two Shades of Black           17999   

     discounted_price  ratings  rating_count  reviews  memory  storage  \
0               14499      4.5         87331     6044     4.0    128.0   
1               47199      4.6        184191    10818     NaN    128.0   
2               15999      4.4         51365     3750     6.0    128.0   
3               11999      4.2         53448     4185     4.0     64.0   
4                9299      NaN        187787    12084     4.0     64.0   
..                ...      ...           ...      ...     ...      ...   
831             35999      4.4          4430      628     NaN    256.0   
832             35999      4.4          4430      628     NaN    256.0   
833             31999      4.4         18194     2478     8.0    128.0   
834             41999      4.4          8689     1232     8.0    128.0   
835             17999      4.4        350744    33202     6.0     64.0   

                    processor               rear_camera      front_camera  \
0     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
1               A Bionic Chip              12MP + 12MP               12MP   
2     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
3      Mediatek Dimensity 700               50MP + 2MP                8MP   
4     Qualcomm Snapdragon 680               50MP + 2MP                5MP   
..                        ...                       ...               ...   
831   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
832   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
833   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
834                       NaN  64MP + 12MP + 8MP + 2MP    32MP + 8MP Dual   
835  Qualcomm Snapdragon 720G   48MP + 8MP + 5MP + 2MP               16MP   

     display_size  battery_capacity     battery_type  total        low  
0           16.36            5000.0          Lithium  34489  14.449206  
1           15.49               NaN              NaN  96099  17.026345  
2           16.36            5000.0          Lithium  36989  13.697333  
3           16.71            5000.0  Lithium Polymer  27998  12.771326  
4           17.02            6000.0  Lithium Polymer  24298  15.540136  
..            ...               ...              ...    ...        ...  
831         16.81            5000.0              NaN  74998   7.054140  
832         16.81            5000.0              NaN  74998   7.054140  
833         16.81            5000.0              NaN  66998   7.342211  
834         16.36            4200.0              NaN  59998   7.052760  
835         16.94            5000.0          Lithium  35998  10.563942  

[836 rows x 18 columns]

#### Displayed the variables name of the dataset in the Lowercase

In [16]:
sample.sort_values(by=['total','low'],ascending=[True,False])

brand                         model          colour  original_price  \
446     ITEL                    ITEL A23S       Ocean Blue            4899   
488     ITEL                    ITEL A23S         Sky Cyan            4899   
505     ITEL                     ITEL A49   Crystal Purple            5799   
745     ITEL                     ITEL A27        Deep Grey            6999   
744     ITEL                     ITEL A27     Crystal Blue            6999   
..       ...                           ...             ...             ...   
764    APPLE          APPLE IPHONE 14 PRO      Space Black          139900   
789    APPLE          APPLE IPHONE 14 PRO      Deep Purple          139900   
665    APPLE          APPLE IPHONE 13 PRO           Silver          149900   
771  SAMSUNG  SAMSUNG GALAXY S23 ULTRA 5G    Phantom Black          149999   
520    APPLE          APPLE IPHONE 14 PRO      Space Black          159900   

     discounted_price  ratings  rating_count  reviews  memory  storage  \
446              4699      3.5            92        6     2.0     32.0   
488              4699      3.5            92        6     2.0     32.0   
505              5699      4.2           587       31     2.0     32.0   
745              5149      4.0           807       55     2.0     32.0   
744              5175      4.0           807       55     2.0     32.0   
..                ...      ...           ...      ...     ...      ...   
764            132999      4.7           756       63     NaN    256.0   
789            132999      4.7           756       63     NaN    256.0   
665            122999      4.6          1421      121     NaN    512.0   
771            124999      3.6            33        3     NaN    256.0   
520            152999      4.7           756       63     NaN    512.0   

                       processor                  rear_camera front_camera  \
446                          NaN                          2MP          NaN   
488                          NaN                          2MP          NaN   
505                    Quad Core                         5MP           5MP   
745                 Unisoc 9832E                         2MP           2MP   
744                 Unisoc 9832E                         2MP           2MP   
..                           ...                          ...          ...   
764         A Bionic Chip,  Core   48MP + 12MP + 12MP + 12MP          12MP   
789         A Bionic Chip,  Core   48MP + 12MP + 12MP + 12MP          12MP   
665                A Bionic Chip          12MP + 12MP + 12MP          12MP   
771  Qualcomm Snapdragon 8 Gen 2  200MP + 10MP + 12MP + 10MP          12MP   
520         A Bionic Chip,  Core   48MP + 12MP + 12MP + 12MP          12MP   

     display_size  battery_capacity     battery_type   total        low  
446         12.70            3020.0              NaN    9598  15.333333  
488         12.70            3020.0              NaN    9598  15.333333  
505         16.76            4000.0  Lithium Polymer   11498  18.935484  
745         13.84            4000.0      Lithium Ion   12148  14.672727  
744         13.84            4000.0      Lithium Ion   12174  14.672727  
..            ...               ...              ...     ...        ...  
764         15.49               NaN              NaN  272899  12.000000  
789         15.49               NaN              NaN  272899  12.000000  
665         15.49               NaN              NaN  272899  11.743802  
771         17.27            5000.0      Lithium Ion  274998  11.000000  
520         15.49               NaN              NaN  312899  12.000000  

[836 rows x 18 columns]

#### In this we have sorted two variables 'Total' and 'Low' and then we have arranged the variables Total in the ascending order

In [17]:
sample.sort_values(by='total', ascending=True)

brand                         model          colour  original_price  \
488     ITEL                    ITEL A23S         Sky Cyan            4899   
446     ITEL                    ITEL A23S       Ocean Blue            4899   
505     ITEL                     ITEL A49   Crystal Purple            5799   
745     ITEL                     ITEL A27        Deep Grey            6999   
744     ITEL                     ITEL A27     Crystal Blue            6999   
..       ...                           ...             ...             ...   
764    APPLE          APPLE IPHONE 14 PRO      Space Black          139900   
665    APPLE          APPLE IPHONE 13 PRO           Silver          149900   
789    APPLE          APPLE IPHONE 14 PRO      Deep Purple          139900   
771  SAMSUNG  SAMSUNG GALAXY S23 ULTRA 5G    Phantom Black          149999   
520    APPLE          APPLE IPHONE 14 PRO      Space Black          159900   

     discounted_price  ratings  rating_count  reviews  memory  storage  \
488              4699      3.5            92        6     2.0     32.0   
446              4699      3.5            92        6     2.0     32.0   
505              5699      4.2           587       31     2.0     32.0   
745              5149      4.0           807       55     2.0     32.0   
744              5175      4.0           807       55     2.0     32.0   
..                ...      ...           ...      ...     ...      ...   
764            132999      4.7           756       63     NaN    256.0   
665            122999      4.6          1421      121     NaN    512.0   
789            132999      4.7           756       63     NaN    256.0   
771            124999      3.6            33        3     NaN    256.0   
520            152999      4.7           756       63     NaN    512.0   

                       processor                  rear_camera front_camera  \
488                          NaN                          2MP          NaN   
446                          NaN                          2MP          NaN   
505                    Quad Core                         5MP           5MP   
745                 Unisoc 9832E                         2MP           2MP   
744                 Unisoc 9832E                         2MP           2MP   
..                           ...                          ...          ...   
764         A Bionic Chip,  Core   48MP + 12MP + 12MP + 12MP          12MP   
665                A Bionic Chip          12MP + 12MP + 12MP          12MP   
789         A Bionic Chip,  Core   48MP + 12MP + 12MP + 12MP          12MP   
771  Qualcomm Snapdragon 8 Gen 2  200MP + 10MP + 12MP + 10MP          12MP   
520         A Bionic Chip,  Core   48MP + 12MP + 12MP + 12MP          12MP   

     display_size  battery_capacity     battery_type   total        low  
488         12.70            3020.0              NaN    9598  15.333333  
446         12.70            3020.0              NaN    9598  15.333333  
505         16.76            4000.0  Lithium Polymer   11498  18.935484  
745         13.84            4000.0      Lithium Ion   12148  14.672727  
744         13.84            4000.0      Lithium Ion   12174  14.672727  
..            ...               ...              ...     ...        ...  
764         15.49               NaN              NaN  272899  12.000000  
665         15.49               NaN              NaN  272899  11.743802  
789         15.49               NaN              NaN  272899  12.000000  
771         17.27            5000.0      Lithium Ion  274998  11.000000  
520         15.49               NaN              NaN  312899  12.000000  

[836 rows x 18 columns]

#### Sorted the variables named 'Total' and arranged its values in ascending order 

In [18]:
sample.sort_values(by='total')

brand                         model          colour  original_price  \
488     ITEL                    ITEL A23S         Sky Cyan            4899   
446     ITEL                    ITEL A23S       Ocean Blue            4899   
505     ITEL                     ITEL A49   Crystal Purple            5799   
745     ITEL                     ITEL A27        Deep Grey            6999   
744     ITEL                     ITEL A27     Crystal Blue            6999   
..       ...                           ...             ...             ...   
764    APPLE          APPLE IPHONE 14 PRO      Space Black          139900   
665    APPLE          APPLE IPHONE 13 PRO           Silver          149900   
789    APPLE          APPLE IPHONE 14 PRO      Deep Purple          139900   
771  SAMSUNG  SAMSUNG GALAXY S23 ULTRA 5G    Phantom Black          149999   
520    APPLE          APPLE IPHONE 14 PRO      Space Black          159900   

     discounted_price  ratings  rating_count  reviews  memory  storage  \
488              4699      3.5            92        6     2.0     32.0   
446              4699      3.5            92        6     2.0     32.0   
505              5699      4.2           587       31     2.0     32.0   
745              5149      4.0           807       55     2.0     32.0   
744              5175      4.0           807       55     2.0     32.0   
..                ...      ...           ...      ...     ...      ...   
764            132999      4.7           756       63     NaN    256.0   
665            122999      4.6          1421      121     NaN    512.0   
789            132999      4.7           756       63     NaN    256.0   
771            124999      3.6            33        3     NaN    256.0   
520            152999      4.7           756       63     NaN    512.0   

                       processor                  rear_camera front_camera  \
488                          NaN                          2MP          NaN   
446                          NaN                          2MP          NaN   
505                    Quad Core                         5MP           5MP   
745                 Unisoc 9832E                         2MP           2MP   
744                 Unisoc 9832E                         2MP           2MP   
..                           ...                          ...          ...   
764         A Bionic Chip,  Core   48MP + 12MP + 12MP + 12MP          12MP   
665                A Bionic Chip          12MP + 12MP + 12MP          12MP   
789         A Bionic Chip,  Core   48MP + 12MP + 12MP + 12MP          12MP   
771  Qualcomm Snapdragon 8 Gen 2  200MP + 10MP + 12MP + 10MP          12MP   
520         A Bionic Chip,  Core   48MP + 12MP + 12MP + 12MP          12MP   

     display_size  battery_capacity     battery_type   total        low  
488         12.70            3020.0              NaN    9598  15.333333  
446         12.70            3020.0              NaN    9598  15.333333  
505         16.76            4000.0  Lithium Polymer   11498  18.935484  
745         13.84            4000.0      Lithium Ion   12148  14.672727  
744         13.84            4000.0      Lithium Ion   12174  14.672727  
..            ...               ...              ...     ...        ...  
764         15.49               NaN              NaN  272899  12.000000  
665         15.49               NaN              NaN  272899  11.743802  
789         15.49               NaN              NaN  272899  12.000000  
771         17.27            5000.0      Lithium Ion  274998  11.000000  
520         15.49               NaN              NaN  312899  12.000000  

[836 rows x 18 columns]

#### We have sorted the variable named 'Total'

In [19]:
sample.drop(['low'],axis=1,inplace=True)

In [20]:
sample

brand             model               colour  original_price  \
0      VIVO      VIVO T1 44W            Starry Sky           19990   
1     APPLE  APPLE IPHONE 11                 White           48900   
2      VIVO      VIVO T1 44W       Midnight Galaxy           20990   
3    XIAOMI       POCO M4 5G           Power Black           15999   
4    XIAOMI         REDMI 10       Caribbean Green           14999   
..      ...               ...                  ...             ...   
831  REALME  REALME GT NEO 2              NEO Blue           38999   
832  REALME  REALME GT NEO 2             NEO Black           38999   
833  REALME  REALME GT NEO 2             NEO Black           34999   
834  REALME   REALME X50 PRO              Rust Red           17999   
835  XIAOMI      POCO M2 PRO   Two Shades of Black           17999   

     discounted_price  ratings  rating_count  reviews  memory  storage  \
0               14499      4.5         87331     6044     4.0    128.0   
1               47199      4.6        184191    10818     NaN    128.0   
2               15999      4.4         51365     3750     6.0    128.0   
3               11999      4.2         53448     4185     4.0     64.0   
4                9299      NaN        187787    12084     4.0     64.0   
..                ...      ...           ...      ...     ...      ...   
831             35999      4.4          4430      628     NaN    256.0   
832             35999      4.4          4430      628     NaN    256.0   
833             31999      4.4         18194     2478     8.0    128.0   
834             41999      4.4          8689     1232     8.0    128.0   
835             17999      4.4        350744    33202     6.0     64.0   

                    processor               rear_camera      front_camera  \
0     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
1               A Bionic Chip              12MP + 12MP               12MP   
2     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
3      Mediatek Dimensity 700               50MP + 2MP                8MP   
4     Qualcomm Snapdragon 680               50MP + 2MP                5MP   
..                        ...                       ...               ...   
831   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
832   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
833   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
834                       NaN  64MP + 12MP + 8MP + 2MP    32MP + 8MP Dual   
835  Qualcomm Snapdragon 720G   48MP + 8MP + 5MP + 2MP               16MP   

     display_size  battery_capacity     battery_type  total  
0           16.36            5000.0          Lithium  34489  
1           15.49               NaN              NaN  96099  
2           16.36            5000.0          Lithium  36989  
3           16.71            5000.0  Lithium Polymer  27998  
4           17.02            6000.0  Lithium Polymer  24298  
..            ...               ...              ...    ...  
831         16.81            5000.0              NaN  74998  
832         16.81            5000.0              NaN  74998  
833         16.81            5000.0              NaN  66998  
834         16.36            4200.0              NaN  59998  
835         16.94            5000.0          Lithium  35998  

[836 rows x 17 columns]

#### We have dropped the variables named 'low' and inplace function

In [21]:
sample.groupby('brand').count()

model  colour  original_price  discounted_price  ratings  \
brand                                                                
APPLE        58      53              58                58       58   
GOOGLE        4       4               4                 4        4   
I KALL        4       4               4                 4        4   
INFINIX      71      71              71                71       71   
IQOO         18      18              18                18       18   
ITEL         18      18              18                18       18   
LAVA          6       6               6                 6        6   
MARQ          1       1               1                 1        1   
MICROMAX      3       3               3                 3        3   
MOTOROLA     53      53              53                53       53   
NOKIA         5       5               5                 5        5   
NOTHING       5       5               5                 5        5   
ONEPLUS       9       9               9                 9        9   
OPPO         56      56              56                56       56   
REALME      144     144             144               144      144   
SAMSUNG     111     111             111               111      111   
TECNO        28      28              28                28       28   
VIVO         63      63              63                63       63   
XIAOMI      179     179             179               179      178   

          rating_count  reviews  memory  storage  processor  rear_camera  \
brand                                                                      
APPLE               58       58       0       53         58           58   
GOOGLE               4        4       4        4          4            4   
I KALL               4        4       4        4          2            4   
INFINIX             71       71      71       71         71           71   
IQOO                18       18      17       14          1           18   
ITEL                18       18      18       18         11           18   
LAVA                 6        6       6        6          5            6   
MARQ                 1        1       1        0          1            1   
MICROMAX             3        3       3        3          3            3   
MOTOROLA            53       53      53       53         53           53   
NOKIA                5        5       5        5          5            5   
NOTHING              5        5       3        0          5            5   
ONEPLUS              9        9       7        9          0            9   
OPPO                56       56      55       56         45           56   
REALME             144      144     136      143        134          144   
SAMSUNG            111      111     110      111         82          111   
TECNO               28       28      28       28         24           28   
VIVO                63       63      62       63         61           63   
XIAOMI             179      179     174      179        133          179   

          front_camera  display_size  battery_capacity  battery_type  total  
brand                                                                        
APPLE               58            58                 0             0     58  
GOOGLE               4             4                 4             0      4  
I KALL               2             4                 4             0      4  
INFINIX             71            71                71            71     71  
IQOO                 7            18                18             0     18  
ITEL                10            18                18            10     18  
LAVA                 4             6                 6             5      6  
MARQ                 1             1                 1             1      1  
MICROMAX             3             3                 3             1      3  
MOTOROLA            53            53                53            32     53  
NOKIA 

#### By performing groupby function on post variable and used count function it is displays us this variable takes place in 835 observations

In [22]:
sample=pd.read_csv("D:/flipkart_smartphones.csv")

In [23]:
sample

brand             model               colour  original_price  \
0      VIVO      VIVO T1 44W            Starry Sky           19990   
1     APPLE  APPLE IPHONE 11                 White           48900   
2      VIVO      VIVO T1 44W       Midnight Galaxy           20990   
3    XIAOMI       POCO M4 5G           Power Black           15999   
4    XIAOMI         REDMI 10       Caribbean Green           14999   
..      ...               ...                  ...             ...   
831  REALME  REALME GT NEO 2              NEO Blue           38999   
832  REALME  REALME GT NEO 2             NEO Black           38999   
833  REALME  REALME GT NEO 2             NEO Black           34999   
834  REALME   REALME X50 PRO              Rust Red           17999   
835  XIAOMI      POCO M2 PRO   Two Shades of Black           17999   

     discounted_price  ratings  rating_count  reviews  memory  storage  \
0               14499      4.5         87331     6044     4.0    128.0   
1               47199      4.6        184191    10818     NaN    128.0   
2               15999      4.4         51365     3750     6.0    128.0   
3               11999      4.2         53448     4185     4.0     64.0   
4                9299      NaN        187787    12084     4.0     64.0   
..                ...      ...           ...      ...     ...      ...   
831             35999      4.4          4430      628     NaN    256.0   
832             35999      4.4          4430      628     NaN    256.0   
833             31999      4.4         18194     2478     8.0    128.0   
834             41999      4.4          8689     1232     8.0    128.0   
835             17999      4.4        350744    33202     6.0     64.0   

                    processor               rear_camera      front_camera  \
0     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
1               A Bionic Chip              12MP + 12MP               12MP   
2     Qualcomm Snapdragon 680         50MP + 2MP + 2MP               16MP   
3      Mediatek Dimensity 700               50MP + 2MP                8MP   
4     Qualcomm Snapdragon 680               50MP + 2MP                5MP   
..                        ...                       ...               ...   
831   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
832   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
833   Qualcomm Snapdragon 870         64MP + 8MP + 2MP               16MP   
834                       NaN  64MP + 12MP + 8MP + 2MP    32MP + 8MP Dual   
835  Qualcomm Snapdragon 720G   48MP + 8MP + 5MP + 2MP               16MP   

     display_size  battery_capacity     battery_type  
0           16.36            5000.0          Lithium  
1           15.49               NaN              NaN  
2           16.36            5000.0          Lithium  
3           16.71            5000.0  Lithium Polymer  
4           17.02            6000.0  Lithium Polymer  
..            ...               ...              ...  
831         16.81            5000.0              NaN  
832         16.81            5000.0              NaN  
833         16.81            5000.0              NaN  
834         16.36            4200.0              NaN  
835         16.94            5000.0          Lithium  

[836 rows x 16 columns]

In [24]:
print(sample['ratings'].isnull().sum())

1


#### We have found out that there are one null values presesnt in the 'Ratings' variable.

In [25]:
print(sample.columns.values)

['brand' 'model' 'colour' 'original_price' 'discounted_price' 'ratings'
 'rating_count' 'reviews' 'memory' 'storage' 'processor' 'rear_camera'
 'front_camera' 'display_size' 'battery_capacity' 'battery_type']


#### Displayed the variables but by using print function